cocos/base/ObjectFactory.h

```diff
@@ -44,12 +44,9 @@ class CC_DLL ObjectFactory
         TInfo();
         TInfo(const std::string& type, Instance ins = nullptr);
         TInfo(const std::string& type, const InstanceFunc& ins = nullptr);
-        TInfo(const TInfo &t);
-        ~TInfo();
-        TInfo& operator= (const TInfo &t);
         std::string _class;
-        Instance _fun;
-        InstanceFunc _func;
+        Instance _fun = nullptr;
+        InstanceFunc _func = nullptr;
```

Comment:
This is initialisation is redundant. std::function default constructor will do this

Reply:
There is clang-tidy option for this, https://releases.llvm.org/7.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability-redundant-member-init.html. We haven't enabled it yet.

I tried with a simple test. It failed to detect `std::function<T> = nullptr`.